### ISSUE TITLE
Julia-based images

### ISSUE THREAD

**Issue by Robinlovelace** [geocompx/docker#69]
- [ ] Base image with core spatial deps installed
- [ ] Addition to pixi-rpy with julia, pixi-rpyjl?

**Comment by asinghvi17** [comment]
Looks like there are still a few problems if you manage the Julia version via pixi:
https://discourse.julialang.org/t/managing-julia-versions-using-pixi/116165/16

**Comment by Robinlovelace** [comment]
Interesting, seems there are a good few Julia projects here: https://pixi.sh/v0.16.1/Community/ some of which are by @evetion none of them use Docker or devcontainers as far as I can see though.

**Comment by Robinlovelace** [comment]
Pure Julia version doesn't need to use Pixi. rpyjl version: probably worthwhile but open to ideas!

**Comment by asinghvi17** [comment]
Hmm...a pure Julia image would be pretty easy then!  Is it easy to automatically update?

**Comment by asinghvi17** [comment]
Hmm...a pure Julia image would be pretty easy then!  Is it easy to automatically update?

**Comment by Robinlovelace** [comment]
Yes, if you supply a Dockerfile we can handle the auto-deployment, just a julia.yml file in the .github/workflows folder should do it.

**Comment by evetion** [comment]
> Interesting, seems there are a good few Julia projects here: https://pixi.sh/v0.16.1/Community/ some of which are by @evetion none of them use Docker or devcontainers as far as I can see though.

Correct, pixi is easier to install than docker and it does everything for us, using the tasks feature. The notable exception is testing our QGIS plugin, which is done in Docker.

**Comment by Robinlovelace** [comment]
I wonder to what extent pixi could replace Docker, with possible resource savings.

**Comment by Robinlovelace** [comment]
This is now working!

![image](https://github.com/user-attachments/assets/1e1c70ba-1c6d-45c5-9076-0c65a5f0c8a3)

**Comment by Robinlovelace** [comment]
Maps looking good: 

![image](https://github.com/user-attachments/assets/8537ccab-9b09-4afd-b6ea-0dde282240c4)
